tools/c7n_gcp/c7n_gcp/filters/log_filter.py

```diff
@@ -0,0 +1,133 @@
+                    resource.labels.instance_id={resource[id]} AND
+                    (
+                        jsonPayload.event_subtype:compute.instances.stop OR
+                        jsonPayload.event_subtype:compute.instances.guestTerminate OR
+                        protoPayload.request.@type:type.googleapis.com/compute.instances.stop
+                    )
+                key: filtered_logs
+                value: empty
+
+    """
+
+    schema = type_schema('stackdriver-logs',
+                         rinherit=ValueFilter.schema,
+                         required=['filter'],
+                         filter={'type': 'string'},
+                         filter_days={'type': 'number', 'minimum': 0})
+    schema_alias = True
+
+    def validate(self):
+        if self.data.get('filter_days') < 0:
+            raise FilterValidationError("Filter '{}': invalid filter_days < 0".format(self.type))
+        if not self.data.get('filter'):
+            raise FilterValidationError("Filter '{}': filter field must exists".format(self.type))
+        super(StackdriverLogFilter, self).validate()
+
+    def process(self, resources, event=None):
+        futures = []
+        results = []
+
+        # Process each resource in a separate thread, returning all that pass filter
+        with self.executor_factory(max_workers=3) as worker:
+            for resource_set in chunks(resources, 20):
+                futures.append(worker.submit(self.process_resource_set, resource_set))
+
+            for feature in as_completed(futures):
+                if feature.exception():
+                    self.log.warning("Log filter error: %s" % feature.exception())
+                    continue
+                else:
+                    results.extend(feature.result())
+
+        return super(StackdriverLogFilter, self).process(results, event=None)
+
+    def process_resource_set(self, resources):
+        project_id = local_session(self.manager.source.query.session_factory).get_default_project()
+        # print("Project {}".format(project_id))
+        client = LogClient(project=project_id, _use_grpc=False)
+
+        time_from = datetime.now() - timedelta(days=self.data.get('filter_days', 30))
+
+        for resource in resources:
+            filter_ = self.data.get('filter').format(resource=resource)
+            if time_from:
+                filter_ = "timestamp>={time_from:%Y-%m-%d} AND ({filter_})".format(
+                    time_from=time_from,
+                    filter_=filter_)
+
+            print("Filter: {}".format(filter_))
+            entries = client.list_entries(filter_=filter_)
+
+            def map_entry(entry):
+                json_entry = LogEntry.to_api_repr(entry)
+                json_entry["payload"] = entry.payload
+                return json_entry
+
+            entries = list(map(map_entry, entries))
+
+            # We are modifying resource to save logs in result
+            resource['filtered_logs'] = entries
```

Comment:
@kapilt It have any sense?
Its really usefull..

Reply:
that seems good, when we modify a resource attributes, our standard atm is to use a c7n: prefix on the key, so it can easily be distinguished from native attributes.

Reply:
OOO.. that prefix is cool.. Will add it..